packages/mermaid/src/diagrams/class/classTypes.ts

```diff
@@ -14,6 +16,119 @@ export interface ClassNode {
   tooltip?: string;
 }
 
+export type Visibility = '#' | '+' | '~' | '-' | '';
+export const visibilityValues = ['#', '+', '~', '-', ''];
+
+/**
+ * Parses and stores class diagram member variables/methods.
+ *
+ */
+export class ClassMember {
+  id!: string;
+  cssStyle!: string;
+  memberType!: 'method' | 'attribute';
+  visibility!: Visibility;
+  /**
+   * denote if static or to determine which css class to apply to the node
+   * @defaultValue ''
+   */
+  classifier!: string;
+  /**
+   * parameters for method
+   * @defaultValue ''
+   */
+  parameters!: string;
+  /**
+   * return type for method
+   * @defaultValue ''
+   */
+  returnType!: string;
+
+  constructor(input: string, memberType: 'method' | 'attribute') {
+    this.memberType = memberType;
+    this.visibility = '';
+    this.classifier = '';
+    this.parseMember(input);
+  }
+
+  getDisplayDetails() {
+    let displayText = this.visibility + parseGenericTypes(this.id);
+    if (this.memberType === 'method') {
+      displayText += `(${parseGenericTypes(this.parameters.trim())})`;
+      if (this.returnType) {
+        displayText += ' : ' + parseGenericTypes(this.returnType);
+      }
+    }
+
+    displayText = displayText.trim();
+    const cssStyle = this.parseClassifier();
+
+    return {
+      displayText,
+      cssStyle,
+    };
+  }
+
+  parseMember(input: string) {
```

Comment:
Shouldn't we sanitize the input? It was done before. 

<img width="527" alt="image" src="https://github.com/mermaid-js/mermaid/assets/10703445/c767c722-695b-4eee-a5ab-0c03f630f06c">

Reply:
Good catch.  I meant to move the handling of that to the ClassMember constructor, but obviously forgot.  Added